router-dbusif.c

```diff
@@ -743,8 +743,10 @@ router_dbusif *router_dbusif_init(struct userdata *u, router_init_data_t* init_d
  * @return void
  */
 
-void router_dbusif_done(struct router_dbusif *routerif) {
+void router_dbusif_done(struct userdata *u) {
     ROUTER_FUNCTION_ENTRY;
+    free_routerif(u);
```

Comment:
Actually this is okay. Inside this function we dont free userdata pointer but we free the u->routerif. The cleanup of this file is required, instead of passing the pointer of userdata to dbus for callbacks, we should pass the pointer of routerif.